apps/api/src/instrument-records/instrument-records.service.ts

```diff
@@ -378,6 +405,38 @@ export class InstrumentRecordsService {
     }
   }
 
+  private expandData(
+    data: InstrumentRecordsExport,
+    listEntry: any[],
+    instrument: ScalarInstrument,
+    record: RecordObject
+  ) {
+    if (listEntry.length > 0) {
+      for (const objectEntry of listEntry) {
+        for (const [dataKey, dataValue] of Object.entries(objectEntry as { [key: string]: any })) {
+          const parseResult = $RecordArrayFieldValue.safeParse(dataValue);
+          if (!parseResult.success) return err('Invalid data value');
+          data.push({
+            groupId: record.groupId ?? DEFAULT_GROUP_NAME,
+            instrumentEdition: instrument.internal.edition,
+            instrumentName: instrument.internal.name,
+            measure: dataKey,
+            sessionDate: record.sessionDate,
+            sessionId: record.sessionId,
+            sessionType: record.sessionType,
+            subjectAge: record.subjectAge,
+            subjectId: record.subjectId,
+            subjectSex: record.subjectSex,
+            timestamp: record.timestamp,
+            value: typeof dataValue === 'string' ? dataValue : JSON.stringify(dataValue)
```

Comment:
why does the value need to be stringified? it should already be stringified when you return it via json encoding